### PR TITLE
Remove unused library

### PR DESCRIPTION
**Pull request by BoyaWu10**
Remove RestSharp, which has a component governance alert of current version, and is never used in our repo.

### CODE REVIEW

**Review by moria97 — DISMISSED** [review]

**Review by sowu880 — DISMISSED** [review]

**Comment by moria97** [comment]
@sowu880 
Not related to this pull request but a similar issue.
Seems we are using [SharpCompress](https://github.com/microsoft/FHIR-Converter/pull/287/files#diff-da92ed0455471050324d61267287874dc347d4a1dfc4bf409fa0d828cf99f927L18) to compress layers and SharpZipLib to decompress layers. This introduces extra dependencies and possible incompatibility issues. Can we use a unified compress lib for compress/decompress operations?

**Review by hongyh13 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj [hunk 1]
```diff
@@ -15,7 +15,6 @@
     <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="3.1.9" />
     <PackageReference Include="Microsoft.Extensions.Options" Version="5.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
-    <PackageReference Include="RestSharp" Version="106.11.5" />
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj:18]: maybe we should also remove the dependency from [nuget spec](https://github.com/microsoft/FHIR-Converter/blob/0a7363e512998f69a37d5484c8716ba7b0b6c236/src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.Liquid.Converter.nuspec#L22)?

Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj:18]: Yes! It is removed in the update.

**Review by moria97 — APPROVED** [review]

**Review by qiwjin — APPROVED** [review]

**Pull request merged**